data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -181,8 +189,12 @@ private void doInitializeInternal() throws IOException {
 
     bulkRequestSupplier = () -> new JavaClientAccumulatingBulkRequest(new BulkRequest.Builder());
     final int maxRetries = openSearchSinkConfig.getRetryConfiguration().getMaxRetries();
+    final OpenSearchClient filteringOpenSearchClient = openSearchClient.withTransportOptions(
+            TransportOptions.builder()
```

Comment:
If these options are always fixed, why not use them in `doInitializeInternal()`?

Reply:
The OpenSearch client has contracts on the data returned for deserialization. We can't filter out the required fields for given calls. The OpenSearch client created in `doInitializeInternal` is used for more than just bulk requests so this filter path would break its contracts

Reply:
Is there a path where `opensearchClient` without filtering is used?

Reply:
Discussed on phone. Resolved.